beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BeamCommitLogReader.java

```diff
@@ -375,6 +376,12 @@ private Instant getWatermark() {
     if (finished) {
       watermark = HIGHEST_INSTANT;
     } else if (eventTime) {
+      if (observer == null) {
+        // Temporary workaround for race condition in UnboundedSourceWrapper.
+        log.warn(
+          "Call to getWatermark() before start(). This breaks UnboundedSource.Reader contract.");
+        return LOWEST_INSTANT;
```

Comment:
I know it's same but maybe we should use `Watermarks.MIN_WATERMARK` and `Watermarks.MAX_WATERMARK` on line 377. WDYT?

Otherwise LGTM as workaround

Reply:
Och, sorry It's Instant....